File: 3788-maximum-unique-subarray-sum-after-deletion/maximum-unique-subarray-sum-after-deletion.cpp
class Solution {
public:
    int maxSum(vector<int>& nums) {
        set<int>s;
        for(int i:nums){
            if(i>0) s.insert(i);
        }

        if(s.empty()) return *max_element(nums.begin(),nums.end());

        return accumulate(s.begin(),s.end(),0);
    }
};

File: 3788-maximum-unique-subarray-sum-after-deletion/maximum-unique-subarray-sum-after-deletion.java
class Solution {
    public int maxSum(int[] nums) {
        
        Set<Integer>posi=new HashSet<>();

        for(int i:nums){
            if(i>0) posi.add(i);
        }
        if(posi.isEmpty()){
            return Arrays.stream(nums).max().getAsInt();
        }

        return posi.stream().mapToInt(Integer::intValue).sum();
    }
}